['text':' Owner(s): ["oncall: jit"]','line_number':1,'multiline':False]
['text':' Make the helper files in test/ importable','line_number':25,'multiline':False]
['text':' Tests for JIT forward hooks and pre-hooks','line_number':38,'multiline':False]
['text':' Hooks can't have the same name as methods.','line_number':109,'multiline':False]
['text':' Test edge case of two hooks sharing name but not python definition','line_number':127,'multiline':False]
['text':' Test that hooks are only invoked when the module is','line_number':167,'multiline':False]
['text':' called directly and not when forward is called.','line_number':168,'multiline':False]
['text':' TODO: add this test back once figured out how to print error msg','line_number':216,'multiline':False]
['text':' Tests if hook error message is printed out if erroring after schema check.','line_number':219,'multiline':False]
['text':' Useful for when user is scripting hooks while not aware of it.','line_number':220,'multiline':False]
['text':' out of bounds tuple range','line_number':225,'multiline':False]
['text':' correct signature: pre_hook_c(self, input: Tuple[str])','line_number':237,'multiline':False]
['text':' expecting Tuple[str], str, or None','line_number':275,'multiline':False]
['text':' expecting Tuple[str], str, or None','line_number':286,'multiline':False]
['text':' doesn't work with eager, inner tuple lost','line_number':299,'multiline':False]
['text':' correct signature:','line_number':313,'multiline':False]
['text':'   def forward_hook(self, input: Tuple[str], output: str)','line_number':314,'multiline':False]
